### PR TITLE
feat: implement pubsub_system_test example

### LINKED ISSUE #103: Implement example: functions_pubsub_system_test
Line numbers are approximate:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/helloworld/hello_pubsub_system_test.go#L18-L73

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/helloworld/sample_pubsub_test_system.py#L15-L62

### PR DESCRIPTION
**Pull request by coryan**
Adds an example showing how to perform system tests for Pub/Sub
triggered functions.

Fixes #103

### CODE REVIEW

File: examples/site/hello_world_pubsub/hello_world_pubsub.cc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // Copyright 2020 Google LLC
+
```
Comment (devjgm) [inline examples/site/hello_world_pubsub/hello_world_pubsub.cc:2]: Is this blank line intentional?

File: examples/site/testing_pubsub/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+We will use this function throughput this guide:
+
+[/examples/site/hello_world_pubsub/hello_world_pubsub.cc]
+```cc
+namespace gcf = ::google::cloud::functions;
+
+// Use Boost.Archive to decode Pub/Sub message payload
+std::string decode_base64(std::string const& base64);
+
+void hello_world_pubsub(gcf::CloudEvent event) {  // NOLINT
```
Comment (devjgm) [inline examples/site/testing_pubsub/README.md:28]: Not sure whether `NOLINT` comments are helpful or needed in docs like this. Up to you.

**Review by devjgm — APPROVED** [review]

File: examples/site/hello_world_pubsub/hello_world_pubsub.cc (hunk 1 above)
Comment (coryan) [reply examples/site/hello_world_pubsub/hello_world_pubsub.cc]: Nope, fixed.

File: examples/site/testing_pubsub/README.md (hunk 2 above)
Comment (coryan) [reply examples/site/testing_pubsub/README.md]: Removed, thanks.

**Pull request merged**